pkg/source/source.go

```diff
@@ -41,54 +42,98 @@ import (
 // * Use Channel for events originating outside the cluster (e.g. GitHub Webhook callback, Polling external urls).
 //
 // Users may build their own Source implementations.
-type Source interface {
+type Source TypedSource[reconcile.Request]
```

Comment:
```suggestion
type Source = TypedSource[reconcile.Request]
```
Maybe?

See: https://github.com/kubernetes-sigs/cluster-api/pull/10803/files#r1656585773

Reply:
Did a quick test. This change fixes the issue in CAPI

Reply:
Great catch, not sure how I missed this